src/client/compound/compound-bridge-data.ts

```diff
@@ -0,0 +1,78 @@
+import { EthAddress } from "@aztec/barretenberg/address";
+import { AssetValue } from "@aztec/barretenberg/asset";
+import { EthereumProvider } from "@aztec/barretenberg/blockchain";
+import { Web3Provider } from "@ethersproject/providers";
+import "isomorphic-fetch";
+import { ICERC20__factory, ICompoundERC4626__factory, IERC20__factory } from "../../../typechain-types";
+import { createWeb3Provider } from "../aztec/provider";
+import { AztecAsset, AztecAssetType } from "../bridge-data";
+
+import { ERC4626BridgeData } from "../erc4626/erc4626-bridge-data";
+
+export class CompoundBridgeData extends ERC4626BridgeData {
+  protected constructor(ethersProvider: Web3Provider) {
+    super(ethersProvider);
+  }
+
+  static create(provider: EthereumProvider) {
+    const ethersProvider = createWeb3Provider(provider);
+    return new CompoundBridgeData(ethersProvider);
+  }
+
+  async getAPR(yieldAsset: AztecAsset): Promise<number> {
+    // Not taking into account how the deposited funds will change the yield
+    // The approximate number of blocks per year that is assumed by the interest rate model
+
+    const blocksPerYear = 2102400;
+
+    // To get the cToken first call cToken on the wrapper contract
+    const cTokenAddress = await this.getCToken(yieldAsset.erc20Address);
+    const cToken = ICERC20__factory.connect(cTokenAddress.toString(), this.ethersProvider);
+
+    const supplyRatePerBlock = await cToken.supplyRatePerBlock();
+    return supplyRatePerBlock.mul(blocksPerYear).toNumber() / 10 ** 16;
+  }
+
+  async getMarketSize(
```

Comment:
Unsure if this is the way we want to compute it. As Compound is a lending/borrowing market, if it is fully utilised, the market size from the balance of the cToken would be 0, where the more "correct" might be the total supply of the wrappers instead.

Reply:
Good point. In AAVE I am fetching totalLiquidity which should correspond to the value we have here. I will just use the wrapper `totalSupply` value and move the function to ERC4626, ok? If you agree I would merge this and then do a separate PR with the change.

Reply:
I could return the totalAssets value from ERC4626 but the think is that the value is wrapper specific while users are probably thinking about the protocol as a whole.
<img width="566" alt="image" src="https://user-images.githubusercontent.com/13470840/200340236-4f13f9a8-8756-4074-b9d0-40e89a26d982.png">

Reply:
Addressed in e8c06f26b832a75a3d68a5145ba2e93d72949e60